Tasks/SonarQubePreBuild/task.json

```diff
@@ -1,127 +1,136 @@
+      "groupName": "project"
+    },
+    {
+      "name": "projectName",
+      "type": "string",
+      "label": "Project Name",
+      "required": true,
+      "helpMarkDown": "The SonarQube project name, i.e. sonar.ProjectName",
+      "groupName": "project"
+    },
+    {
+      "name": "projectVersion",
+      "type": "string",
+      "label": "Project Version",
+      "defaultValue": "1.0",
+      "required": true,
+      "helpMarkDown": "The SonarQube project version, i.e. sonar.projectVersion",
+      "groupName": "project"
+    },
+    {
+      "name": "connectedServiceName",
+      "type": "connectedService:Generic",
+      "label": "SonarQube Endpoint",
+      "required": true,
+      "helpMarkDown": "The SonarQube server generic endpoint",
+      "groupName": "serverSettings"
+    },
+    {
+      "name": "dbUrl",
+      "type": "string",
+      "label": "Db Connection String",
+      "required": false,
+      "helpMarkDown": "SonarQube server 5.1 and lower only. The database connection setting, i.e. sonar.jdbc.url. For example jdbc:jtds:sqlserver://localhost/sonar;SelectMethod=Cursor",
+      "groupName": "dbSettings"
+    },
+    {
+      "name": "dbUsername",
+      "type": "string",
+      "label": "Db UserName",
+      "required": false,
+      "helpMarkDown": "SonarQube server 5.1 and lower only The username for the database user, i.e. sonar.jdbc.username.",
+      "groupName": "dbSettings"
+    },
+    {
+      "name": "dbPassword",
+      "type": "string",
+      "label": "Db User Password",
+      "required": false,
+      "helpMarkDown": "SonarQube server 5.1 and lower only. The password for the database user, i.e. sonar.jdbc.password",
+      "groupName": "dbSettings"
+    },
+    {
+      "name": "cmdLineArgs",
+      "type": "string",
+      "label": "Additional Settings",
+      "required": false,
+      "helpMarkDown": "Space separated settings using the format: /d:propertyName=propertyValue. Normal command line escaping rules apply. For more details on the settings please view the [SonarQube docs](http://go.microsoft.com/fwlink/?LinkID=624390).",
+      "groupName": "advanced"
+    },
+    {
+      "name": "configFile",
+      "type": "filePath",
+      "defaultValue": "",
+      "label": "Settings File",
+      "required": false,
+      "helpMarkDown": "You can also specify settings via a configuration file. A template is available [here](http://go.microsoft.com/fwlink/?LinkID=620062)",
+      "groupName": "advanced"
+    },
+    {
+      "name": "breakBuild",
+      "type": "boolean",
+      "defaultValue": "",
+      "label": "Fail the build on quality gate failure (SQ 5.3+)",
+      "required": false,
+      "helpMarkDown": "This option is only available when using a SonarQube server 5.3 or above. It will introduce delays as the build must wait for SonarQube to complete the analysis. [More information](http://go.microsoft.com/fwlink/?LinkId=722407)",
```

Comment:
Currently the URL points to http://www.sonarqube.org/ but I'll change it so it points to the blog post.